airflow-core/src/airflow/jobs/scheduler_job_runner.py

```diff
@@ -270,6 +272,98 @@ def register_signals(self) -> ExitStack:
 
         return resetter
 
+    def _get_team_names_for_dag_ids(
+        self, dag_ids: Collection[str], session: Session
+    ) -> dict[str, str | None]:
+        """
+        Batch query to resolve team names for multiple DAG IDs using the DAG > Bundle > Team relationship chain.
+
+        DAG IDs > DagModel (via dag_id) > DagBundleModel (via bundle_name) > Team
+
+        :param dag_ids: Collection of DAG IDs to resolve team names for
+        :param session: Database session for queries
+        :return: Dictionary mapping dag_id -> team_name (None if no team found)
+        """
+        if not dag_ids:
+            return {}
+
+        try:
+            # Query all team names for the given DAG IDs in a single query
+            query_results = session.execute(
+                select(DagModel.dag_id, Team.name)
+                .join(DagBundleModel.teams)  # Join Team to DagBundleModel via association table
+                .join(
+                    DagModel, DagModel.bundle_name == DagBundleModel.name
+                )  # Join DagBundleModel to DagModel
+                .where(DagModel.dag_id.in_(dag_ids))
+            ).all()
+
+            # Create mapping from results
+            dag_id_to_team_name = {dag_id: team_name for dag_id, team_name in query_results}
+
+            # Ensure all requested dag_ids are in the result (with None for those not found)
+            result = {dag_id: dag_id_to_team_name.get(dag_id) for dag_id in dag_ids}
+
+            self.log.debug(
+                "Resolved team names for %d DAGs: %s",
+                len([team for team in result.values() if team is not None]),
+                {dag_id: team for dag_id, team in result.items()},
+            )
+
+            return result
+
+        except Exception:
+            # Log the error, explicitly don't fail the scheduling loop
+            self.log.exception("Failed to resolve team names for DAG IDs: %s", list(dag_ids))
```

Comment:
I would also include the exception itself in the log message. `log.exception` does it pretty well, it prints out the stack trace

Reply:
Oh! Looking at other examples you might be right. I was convinced we had to pass manually the exception to the log function so that it can print the stacktrace but now you're saying this, you're probably right. If you could double check, that'd be awesome though

Reply:
Nice :) LGTM (I cannot approve a closed PR)